src/screens/home/HomeScreen.tsx

```diff
@@ -45,17 +40,19 @@ export const HomeScreen = ({
         navigation={navigation}
         route={route}
         setBarStyle={setBarStyle}
-        setLauncherContext={setLauncherContext}
+        setLauncherContext={trySetLauncherContext}
       />
 
-      {isLauncherReady(launcherContext) && isClientReady(launcherClient) && (
+      {canDisplayLauncher() && (
         <LauncherView
           launcherClient={launcherClient}
           launcherContext={launcherContext.value}
-          retry={(): void => setLauncherContext({ state: 'default' })}
+          retry={resetLauncherContext}
           setLauncherContext={setLauncherContext}
         />
       )}
+
+      {LauncherDialog}
```

Comment:
nit: with that name we expect to potentially see something different than an error, with maybe logic inside. Should we rename it LauncherErrorDialog?

Reply:
Yes, that was my intent actually, the HomeScreen doesn't really want or need to know what's going to display in that LauncherDialog props. One thing is sure it will be a Dialog, but not necessarily linked to an error.